hw3/2_MiniLogo.hs

```diff
@@ -0,0 +1,46 @@
+-- Mini Logo
+
+data Cmd
+  = Pen Mode
+  | MoveTo Int Int
+  | Sequ Cmd Cmd
+  deriving (Show)
+
+data Mode = Up | Down deriving (Show)
+
+type State = (Mode, Int, Int)
+
+type Line = (Int, Int, Int, Int)
+
+type Lines = [Line]
+
+semL :: Cmd -> State -> (State, Lines)
+semL (Pen m) (_, x, y) = ((m, x, y), [])
+semL (MoveTo a b) (Down, x, y) = ((Down, a, b), [(x, y, a, b)])
+semL (MoveTo a b) (Up, x, y) = ((Down, x, y), [])
```

Comment:
If I move the pen while it's up, why would it then become down after I finish the move? Shouldn't it stay up?

Reply:
good catch! I'll change that.